components/Column.vue

```diff
@@ -1,15 +1,22 @@
 <script setup lang="ts">
 import { VueDraggable } from 'vue-draggable-plus'
-import type { Issue, IssueStatus } from '../constants/issues'
+import type { Issue } from '../types/issues'
+import type { ColumnTitle } from '~/constants/columns'
 
-const props = defineProps<{ title: IssueStatus; issues: Issue[] }>()
-const issuesModel = ref(structuredClone(toRaw(props)).issues) // create new ref with the same contents
+const props = defineProps<{ title: ColumnTitle; issues: Issue[] }>()
 
-const STATUS_TO_ICON_MAP = {
-  todo: { name: 'bx:circle', class: 'text-rad-foreground-contrast' },
-  doing: { name: 'bx:adjust', class: 'text-rad-foreground-yellow' },
-  done: { name: 'bx:bxs-circle', class: 'text-rad-foreground-success' },
-} satisfies Record<IssueStatus, { name: string; class: string }>
+const issuesModel = ref<Issue[]>([])
+
+watchEffect(() => {
+  issuesModel.value = [...unref(toRef(props, 'issues'))] // "clone" issues prop
+})
```

Comment:
Here I wanted the `issuesModel` to be overwritten by `props.issues` if it got updated. This didn't seem to happen with `structuredClone`.

Happy to change this if you have any suggestions.

Reply:
We'll definitely need to have a better look into this later after the refactorings of all other discussions settle. This is super odd, especially `unref()` after `toRef` (did you mean to use `toRefs()` pehaps?) and even more so inside a watchEffect :grimacing: . This may well be forcing expensive re-triggering of the watchEffect and/or needless re-renders.

I'll follow up on this eventually if you haven't by then.

Reply:
I've simplified it a bit. Please have a look